src/core/stores/stores.ts

```diff
@@ -0,0 +1,25 @@
+import { create } from 'zustand';
+import type { Character } from '../interfaces/interface';
+
+interface ICharacter {
+  characters: Character[];
+  selectCharacters: (newCharacter: Character) => void;
+  unselectCharacter: (id: number) => void;
+  unselectAllCharacters: () => void;
+}
+
+export const useCharactersStore = create<ICharacter>()((set) => ({
+  characters: [],
+  selectCharacters: (newCharacter: Character) =>
+    set((state) => ({
+      characters: [...state.characters, newCharacter],
+    })),
+  unselectCharacter: (id: number) =>
+    set((state) => ({
+      characters: state.characters.filter((character) => character.id !== id),
+    })),
+  unselectAllCharacters: () =>
+    set(() => ({
+      characters: [],
+    })),
```

Comment:
As I see it this store is used for selecting/deselecting characters. Seems it is better to store only id's of selected items. No need to store all the other data if you don't use it 
```suggestion
  selectedCharacters: [],
  selectCharacters: (charId: number) =>
    set((state) => ({
      selectedCharacters: [...state.selectedCharacters, charId],
    })),
  unselectCharacter: (id: number) =>
    set((state) => ({
      selectedCharacters: state.selectedCharacters.filter((id) => id !== charId),
    })),
  unselectAllCharacters: () =>
    set(() => ({
      characters: [],
    })),
```

Reply:
Thanks, this is good idea. But, I use select characters data for download information in csv